src/app/api/products/route.ts

```diff
@@ -0,0 +1,48 @@
+import { NextRequest, NextResponse } from "next/server";
+import { Product } from "@/types/types";
+import { executeSOQLQuery } from "@/lib/salesforce/soqlQuery";
+
+interface QueryResponse<T> {
+    totalSize: number;
+    done: boolean;
+    records: Array<T>;
+}
+
+interface PricebookEntry {
+    Id: string;
+    Name: string;
+    Product2Id: string;
+    Product2: Product2;
+}
+
+interface Product2 {
+    Family: string;
+}
+
+export async function GET(req: NextRequest) {
+    const query = `
+        SELECT Id, Name, product2id, Product2.Family
+        FROM PricebookEntry
+        WHERE Pricebook2Id = '01si0000002Ip3WAAS'
+    `;
+    try {
+        const data: QueryResponse<PricebookEntry> = await executeSOQLQuery(query);
+        const records = data.records;
+        const groupedByFamily = records.reduce((acc, product) => {
+            const family = product.Product2.Family || 'Uncategorized';
+            if (!acc[family]) {
+                acc[family] = [];
+            }
+            acc[family].push({
+                id: product.Id,
+                name: product.Name,
+                category: family,
+            });
+            return acc;
+        }, {} as Record<string, Product[]>);
+        return NextResponse.json(groupedByFamily, { status: 200 });
```

Comment:
```suggestion
        return NextResponse.json(groupedByFamily, { status: 200 });
```

to sort the categories & products alphabetically, this line should work:
const sortedProducts = Object.keys(groupedByFamily) 
                             .sort((a, b) => a.localeCompare(b)) // sort by category name
                             .map((family: string) => { return {category: family, products: groupedByFamily[family].sort((a, b) => a.name.localeCompare(b.name)) }})

Reply:
since this would return an array of type { category: string; products: Product[] }[], it would require a bit of refactoring on the frontend to deal with the changed type structure